storage.bs

```diff
@@ -436,28 +474,44 @@ evaluating quotas.
 
 
 
-<h2 id=ui-guidelines>User Interface Guidelines</h2>
+<h2 id=management>Management</h2>
+
+<p>Whenever a <a>storage bucket</a> is cleared by the user agent, it must be cleared in its
+entirety. User agents should avoid clearing <a>storage buckets</a> while script that is able to
+access them is running, unless instructed otherwise by the user.
+
+<p>If removal of <a>storage buckets</a> leaves the encompassing <a>storage shelf</a>'s
+<a>bucket map</a> <a for=map lt="is empty">empty</a>, then <a for=map>remove</a> that
+<a>storage shelf</a> and corresponding <a>storage key</a> from the encompassing
+<a for=/>storage shed</a>.
+
+
+<h3 id=storage-pressure>Storage pressure</h3>
+
+<p>A user agent that comes under storage pressure should clear network state and
+<a>local storage buckets</a> whose <a for="local storage bucket">mode</a> is
+"<code>best-effort</code>", ideally prioritizing removal in a manner that least impacts the user.
+
+<p>If a user agent continues to be under storage pressure, then the user agent should inform the
+user and offer a way to clear the remaining <a>local storage buckets</a>, i.e., those whose
+<a for="local storage bucket">mode</a> is "<code>persistent</code>".
+
+<p><a>Session storage buckets</a> must be cleared as <a for=/>browsing sessions</a> are closed.
+
+<p class="note">If the user agent allows revival of <a for=/>browsing sessions</a>, then nothing can
+be cleared until revival is no longer possible.
```

Comment:
I think HTML's

> The lifetime of a browsing context can be unrelated to the lifetime of the actual user agent process itself, as the user agent can support resuming sessions after a restart.

provides extra clarity, whereas this note is not as connected to the user experience of sessions surviving a restart. I'd suggest adding something like

> This means that cessation of the user agent process might not allow clearing session storage, if restarting the user agent revives previously-active browsing sessions.